x-pack/plugin/src/test/resources/rest-api-spec/test/runtime_fields/90_loops.yml

```diff
@@ -96,10 +147,56 @@ setup:
              terms:
                field: lla
 
+---
+"Max chain depth - 5 is allowed":
+  - do:
+      search:
+        index: sensor
+        body:
+          sort: timestamp
+          docvalue_fields: [over_max_depth_1]
+
+  - match: {hits.total.value: 1}
+  - match: {hits.hits.0.fields.over_max_depth_1.0: test}
+
+---
+"Max chain depth - direct references are not counted":
   - do:
-      catch: '/Cyclic dependency detected while resolving runtime fields: lla -> llb -> llc -> loose_loop -> lla/'
       search:
         index: sensor
         body:
           sort: timestamp
-          docvalue_fields: [lla]
+          docvalue_fields: [refs]
+
+  - match: {hits.total.value: 1}
+  - match: {hits.hits.0.fields.refs.0: testtesttesttesta}
+
+---
+"Max chain depth - aggs":
+  - do:
+      catch: '/Field requires resolving too many dependent fields: over_max_depth -> over_max_depth_1 -> over_max_depth_2 -> over_max_depth_3 -> over_max_depth_4 -> over_max_depth_5/'
+      search:
+        index: sensor
+        body:
+          sort: timestamp
+          aggs:
+            loop:
+             terms:
+               field: over_max_depth
+---
+"Max chain depth - sort":
+  - do:
+      catch: '/Field requires resolving too many dependent fields: over_max_depth -> over_max_depth_1 -> over_max_depth_2 -> over_max_depth_3 -> over_max_depth_4 -> over_max_depth_5/'
+      search:
+        index: sensor
+        body:
+          sort: over_max_depth
+---
+"Max chain depth - docvalue_fields":
+  - do:
+      catch: '/Field requires resolving too many dependent fields: over_max_depth -> over_max_depth_1 -> over_max_depth_2 -> over_max_depth_3 -> over_max_depth_4 -> over_max_depth_5/'
+      search:
+        index: sensor
+        body:
+          sort: timestamp
+          docvalue_fields: [over_max_depth]
```

Comment:
Note that there are not tests for queries, on purpose. They need to be fixed :)

Reply:
Yeah, I have this on my list but I keep getting distracted by shiny objects.